File: solution files/Python/part-05/task-01-url-validation/url_validation.py
from urllib.parse import urlparse

ALLOWED_HOSTS = {
    "raw.githubusercontent.com",
    "league-blueprints.internal",
}


def validate_url(url: str) -> None:
    """Raise ValueError if the URL's host is not in the allowlist."""
    parsed = urlparse(url)
    if parsed.scheme not in ("https",):
        raise ValueError(f"URL must use HTTPS. Got: {parsed.scheme!r}")
    if parsed.netloc not in ALLOWED_HOSTS:
        raise ValueError(f"Host '{parsed.netloc}' is not in the allowed list.")


if __name__ == "__main__":
    try:
        validate_url(
            "https://raw.githubusercontent.com/microsoft/CopilotAdventures/main/Data/scrolls.txt"
        )
        print("URL is valid")
    except ValueError as e:
        print(f"Blocked: {e}")

File: solution files/Javascript/part-05/task-01-url-validation/urlValidation.js
const ALLOWED_HOSTS = new Set([
    "raw.githubusercontent.com",
    "league-blueprints.internal",
]);

/**
 * Throw an Error if the URL is not HTTPS or the host is not in the allowlist.
 * @param {string} url
 */
function validateUrl(url) {
    const parsed = new URL(url);
    if (parsed.protocol !== "https:")
        throw new Error(`URL must use HTTPS. Got: '${parsed.protocol}'`);
    if (!ALLOWED_HOSTS.has(parsed.hostname))
        throw new Error(`Host '${parsed.hostname}' is not in the allowed list.`);
}

module.exports = { validateUrl };
